docs/guide_slash_and_interactions.rst

```diff
@@ -0,0 +1,297 @@
+.. Slash Commands and Interactions
+
+.. role:: python(code)
+    :language: python
+
+===============================
+Slash Commands and Interactions
+===============================
+
+This guide is going to cover on how to write a simple slash command into a Red cog.
+This guide will assume that you have a working basic cog.
+If you do not have a basic cog, please refer to the :ref:`getting started <getting-started>` guide.
+It is also adviced to make your self familiar with `Application Commands <https://discord.com/developers/docs/interactions/application-commands>`__ from Discord's documentation. 
+
+---------------
+Getting Started
+---------------
+
+To start off, we will have to import some additional modules to our cog file.
+We will be using the :class:`redbot.core.app_commands` module to create our slash commands.
+Once we have imported the module, we can start creating our slash commands in our cog class.
+For this example we will use a basic hello world command.
+
+.. code-block:: python
+
+    import discord
+
+    from redbot.core import commands, app_commands
+
+    class MyCog(commands.Cog):
+        def __init__(self, bot):
+            self.bot = bot
+
+        @app_commands.command()
+        async def hello(self, interaction: discord.Interaction):
+            await interaction.response.send_message("Hello World!", ephemeral=True)
+
+Go ahead and load your cog. Once it is loaded, we will have to register our slash commands.
+We can do this by using the :ref:`Slash<core-command-slash>` command to manage our slash commands.
+Once you have registered your slash commands, you can test them out by typing ``/hello`` in your server.
+
+----------------------------
+Slash Commands and Arguments
+----------------------------
+
+There is a lot of flexibility when it comes to slash commands.
+Below we will go over some of the different stuff you can do with slash commands.
+
+Decorators
+----------
+Just like with text commands, we can use decorators to modify the behaviour of our slash commands.
+For example, we can use the :func:`app_commands.guild_only` decorator to make our slash command only work in guilds.
+
+.. code-block:: python
+
+    import discord
+
+    from redbot.core import commands, app_commands
+
+    class MyCog(commands.Cog):
+        def __init__(self, bot):
+            self.bot = bot
+
+        @app_commands.command()
+        @app_commands.guild_only()
+        async def hello(self, interaction: discord.Interaction):
+            await interaction.response.send_message("Hello World!", ephemeral=True)
+
+One of the more useful decorators is the :func:`app.commands.choices` decorator.
```

Comment:
I don't think your example uses this decorator, rather it uses the `enum` method of specifying choices. There's like 3 different ways that all work for specifying choices, this is just one of them.